enhancements/cluster-logging/azure_workload_identity_auth.md

```diff
@@ -0,0 +1,264 @@
+
+## Release Sign-off Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+[Azure Monitor Logs](https://learn.microsoft.com/en-us/azure/azure-monitor/logs/data-platform-logs) is a comprehensive service provided by Microsoft Azure that enables the collection, analysis, and actioning of telemetry data across various Azure and on-premises resources.
+
+This proposal enhances the Azure Monitor Logs integration by implementing secure, short-lived authentication with [Microsoft Entra Workload Identity (WID)](https://learn.microsoft.com/en-us/entra/workload-id/workload-identities-overview) through federated tokens. The update will leverage a pending, upstream Vector PR, [azure_logs_ingestion feature](https://github.com/vectordotdev/vector/pull/22912), that will utilize the new Log Ingestion API.
+
+## Motivation
+
+The current Azure Monitor Logs integration relies on long-lived credentials ([shared_key](https://learn.microsoft.com/en-us/previous-versions/azure/azure-monitor/logs/data-collector-api?tabs=powershell#authorization)) and a deprecated [API](https://learn.microsoft.com/en-us/previous-versions/azure/azure-monitor/logs/data-collector-api), which poses potential security risks. By adopting Microsoft Entra Workload Identity (WID), this enhancement ensures secure, short-lived credential access to Azure Monitor Logs while removing reliance on a deprecated, soon to be retired, API.
+
+### User Stories
+
+- As an administrator, I want to be able to forward logs from my OpenShift cluster to Azure Monitor Logs using federated tokens, removing the need for long-lived, static credentials.
+
+### Goals
+
+- Enable Vector's Azure Monitor Logs sink to authenticate with short-lived federated token credentials.
+
+### Non-Goals
+
+- Supporting authentication other than short lived federated token credentials.
+
+## Proposal
+
+To realize the goals of this enhancement:
+
+- Switch over to the new Azure Log Ingestion vector sink when implemented.
+  - See #1 in [implementation details](#implementation-detailsnotesconstraints) section.
+- Update Vector's rust [Azure Identity](https://github.com/Azure/azure-sdk-for-rust/tree/main/sdk/identity/azure_identity) client library to `v0.23.0`.
+  - See #2 in [implementation details](#implementation-detailsnotesconstraints) section.
+- Extend Vector's Azure Log Ingestion sink to accept additional configuration for workload identity authentication.
+- Extend the ClusterLogForwarder’s Azure Monitor integration to support the required fields of the Log Ingestion API, including additional authentication fields for workload identity.
+
+### Workflow Description
+
+The Vector collector will:
+
+1. Determine the authentication type using a configurable field (`credential_kind`).
```

Comment:
Our plan for non-AWS (gcp, azure, alibaba) is to have auth type determined by the type of credentials you are pointing to.   This is how local config and credential files work in both gcp and aws.   Does azure require any 'type' when you auth with their cli?  If not, then I'd suggest we align as closely as possible to our aws and (wip) gcp auth.